presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/PlanPrinter.java

```diff
@@ -1032,6 +1026,7 @@ private void printTableScanInfo(TableScanNode node, int indent)
                 }
             }
 
+            TupleDomain<ColumnHandle> predicate = node.getLayoutConstraint();
```

Comment:
why not to use `TableScanNode::getCurrentConstraint()` here?

Reply:
The current constrain is pruned to only columns included in the output.  Another way to think about it is, why was the original code doing this:
```java
TupleDomain<ColumnHandle> predicate = node.getLayout()
                .map(layoutHandle -> metadata.getLayout(session, layoutHandle))
                .map(TableLayout::getPredicate)
                .orElse(TupleDomain.all());
```
For the explain plan we want to know the Hive partitions that are scanned, and they will be dropped if current constraint unless they are actually output from the scan node.

Reply:
As far I understand this code, current constraint should contain information about all table columns which are filtered, not only columns which are in output. 

For example see a recent change in `IndexSourceNode` which is conceptually very similar to `TableScanNode` -  https://github.com/prestodb/presto/commit/3eac47dfec97401309700871d3b4a79c20090ca5

Can you please point me a place where such current constraint pruning happens? Tomorrow I will try to look into into the code by myself. Today I am just talking out of my head.

Reply:
Notice, once upon a time I thought the same way as you. I wanted to enforce that in `TableScanNode` here: https://github.com/prestodb/presto/pull/9808. But that was wrong.

Reply:
@martint suggested this strategy.  Maybe he knows where the pruning happens.

Reply:
That would be a bug to me.  However, I see that there is no test coverage for that and such pruning may exists accidentally.

Maybe we could introduce some legacy session switch on which we would do `metadata.getLayout` to access the predicate while on default we would use 
 just `TableScanNode::currentConstraint`. That way we would catch the cases where they do not match so we would be able to write tests and fix these bugs. Eventually we would be able to remove that switch. 

How does it sound to you?